internal/crypto/keystores/keystore.go

```diff
@@ -31,32 +34,59 @@ type KeyStore interface {
 	GetKey(id string) ([]byte, error)
 }
 
+// LocalKeyStore is the config value for an on-disk key store
+const LocalKeyStore = "local"
+
 // ErrUnknownKeyID is returned when the Key ID cannot be found by the keystore.
 var ErrUnknownKeyID = errors.New("unknown key id")
 
-// This structure is used by the keystore implementation to manage keys.
 type keysByID map[string][]byte
 
 // NewKeyStoreFromConfig creates an instance of a KeyStore based on the
 // AuthConfig in Minder.
 // Since our only implementation is based on reading from the local disk, do
 // all key loading during construction of the struct.
-// TODO: allow support for multiple keys/algos
-func NewKeyStoreFromConfig(config *serverconfig.AuthConfig) (KeyStore, error) {
-	key, err := config.GetTokenKey()
+func NewKeyStoreFromConfig(config serverconfig.CryptoConfig) (KeyStore, error) {
+	// You might think that a nil value cannot be deserialized to a struct,
+	// yet, here we are.
+	if config.KeyStore.Config == nil {
+		return nil, errors.New("keystore config is missing")
+	}
+
+	// TODO: support other methods in future
+	if config.KeyStore.Type != LocalKeyStore {
+		return nil, fmt.Errorf("unexpected keystore type: %s", config.KeyStore.Type)
+	}
+
+	var keystoreCfg serverconfig.LocalFileKeyStoreConfig
+	err := mapstructure.Decode(config.KeyStore.Config, &keystoreCfg)
```

Comment:
This is because Config is a `map[string]any`, right?

(I'd like to minimize the total area of `mapstructure` in our code, though I'm willing to be convinced that I shouldn't)

Reply:
Yes, it's because the exact type of `Config` is not known until the `Type` field is read.

Perhaps this could be done as part of config parsing, but I felt it would be better to do at the point where we make use of the config.

Reply:
As I said, I'm okay with this pattern -- it would be nice if we could 👍  or 👎  the config at parsing time, but that's already not the case.